docs/Course Wiki/CS Course Offerings/CS374A.md

```diff
@@ -0,0 +1,95 @@
+# CS374A
+
+## Content Covered
+
+### First third (languages)
+
+- Induction, with a focus on application to strings
+- Regular languages
+- Regular expressions
+- Deterministic finite autoamta (DFAs)
+- Nondeterminstic finite automata (NFAs)
+- Regular language transformation via DFA and NFA manipulations
+- Myhill-Nerode theorem and proving irregularity with fooling sets
+- Context free languages and context free grammars (CFGs)
+- Turing machines (only some semesters)
+
+### Middle third (algorithms)
+
+- Divide and conquer algorithms
+- Backtracking algorithms
+- Dynamic programming algorithms
+- Graph algorithms
+- Greedy algorithms
+
+### Final third (complexity)
+
+- Undecidable languages
+- Undecidability of the halting problem
```

Comment:
Add proving undecideability (or undecideability reductions)?

Undecideability isn't covered every semester either iirc. You can switch the order of undecideability and NP-hardness/completeness, because NP is definitely covered every semester.